diff --git a/Makefile b/Makefile
index 878e147..3fd9034 100644
--- a/Makefile
+++ b/Makefile
@@ -196,6 +196,22 @@ else ifeq ($(platform), ps2)
    STATIC_LINKING = 1
    OLD_GCC = 1
 
+# target platform tg5050 trimUI smart pro s
+else ifeq ($(platform), tg5050)
+   TARGET := $(TARGET_NAME)_libretro.so
+   CC = $(CROSS_COMPILE)gcc
+   CXX = $(CROSS_COMPILE)g++
+   AR = $(CROSS_COMPILE)ar
+   fpic := -fPIC
+   SHARED := -shared -Wl,--no-undefined -Wl,--version-script=link.T
+   LDFLAGS += -lrt
+   FLAGS += -fomit-frame-pointer -ffast-math -flto
+   FLAGS += -mcpu=cortex-a55
+   LDFLAGS += -flto
+   ifneq ($(DEBUG),1)
+      LDFLAGS += -s
+   endif
+
 # PSP
 else ifeq ($(platform), psp1)
    TARGET := $(TARGET_NAME)_libretro_$(platform).a
